### PR TITLE
slight refactor

### PR DESCRIPTION
**Pull request by benjaminxscott**
avoids executing the removal logic or printing empty 'removal' log lines w/ 5 players

### CODE REVIEW

**Comment by tobymao** [comment]
this will probably break every 5 player game because now the random seeds have changed

**Comment by benjaminxscott** [comment]
oh does the digest of the engine file get used as part of the prng seed?

**Comment by benjaminxscott** [comment]
Oh would it regenerate current games based on the new `rand` call?

**Comment by tobymao** [comment]
> Oh would it regenerate current games based on the new `rand` call?

yes, every game is purely made up by it's actions. if the number of rands change, that will change the subsequent random decisions in the game, so i'm inclined to not merge this change although I do appreciate your effort in the PR.

**Comment by benjaminxscott** [comment]
Ah, so if we instead check for `remove.nil?` on the print statement, would that be a non-breaking prng seed change? 

(since it's still calling rand() the same number of times, it's just not printing)

**Comment by tobymao** [comment]
yes, if the goal is to simply clean up the log in a 5 player game.

also you can run lint locally

docker-compose exec rack rubocop

pass -e if you want it to auto correct

**Comment by benjaminxscott** [comment]
Ah yep my commits probably look funny, haven't setup a ruby env on my chromebook 
(also i don't have much xp with ruby)

**Comment by tobymao** [comment]
you just need docker, not sure if that can work on a chromebook

File: lib/engine/game/g_1846.rb [hunk 1]
```diff
@@ -134,10 +134,13 @@ def setup
       end
 
       def remove_from_group!(group, entities)
-        remove = group.sort_by { rand }.take([5 - @players.size, 2].min)
-        @log << "Removing #{remove.join(', ')}"
+        removals = group.sort_by { rand }.take([5 - @players.size, 2].min)
```
Comment (tobymao) [inline lib/engine/game/g_1846.rb:137]: this doesn't do what you think it does, sort_by.take returns an array

an empty array is truthy in ruby unlike in python

ruby only false and nil are falsey

you need to do return if removals.empty?

Comment (benjaminxscott) [reply lib/engine/game/g_1846.rb:137]: Ah, thanks !

**Pull request merged**